sdk/core/core-http/lib/policies/atomSerializationPolicy.ts

```diff
@@ -51,14 +52,18 @@ export class AtomSerializationPolicy extends BaseRequestPolicy {
       request.body = serializer.serialize(JSON.parse(request.body));
     }
 
-    let response: HttpOperationResponse = await this._nextPolicy.sendRequest(request);
+    const response: HttpOperationResponse = await this._nextPolicy.sendRequest(request);
 
     try {
       if (response.bodyAsText) {
         response.parsedBody = await deserializeAtomXmlToJson(response.bodyAsText);
       }
     } catch (e) {
-      throw buildAtomError({ code: "ResponseNotInAtomXMLFormat" }, response);
+      const error = new RestError("ResponseNotInAtomXMLFormat", RestError.PARSE_ERROR);
+      error.statusCode = 400;
```

Comment:
What's the reason for setting the status code to 400 instead of what the service returned? If the service gave us a malformed payload they might be sending a different status code as well.

Reply:
We'd still want users to see such scenarios as an erroneous one.
If the service doesn't already mark it as an errored response, we should update it I think.
Updated code to reflect this check so we retain the status code returned by service, if it was a meaningful/appropriate one

Reply:
My concern with using 400 as the fallback status code if service doesnt return any is that we are now saying that this is a case of "Bad Request" which may not be the case. It could be a service error.

Reply:
I would prefer if we follow the model being used in the [deserializationPolicy.ts](https://github.com/Azure/azure-sdk-for-js/blob/%40azure/core-http_1.0.0-preview.3/sdk/core/core-http/lib/policies/deserializationPolicy.ts) file which is the policy used for all our other http request/responses

Reply:
Updated to use the status on response from service, as is the case in the deserializationPolicy https://github.com/Azure/azure-sdk-for-js/blob/%40azure/core-http_1.0.0-preview.3/sdk/core/core-http/lib/policies/deserializationPolicy.ts#L101